arkouda/strings.py

```diff
@@ -678,21 +678,20 @@ def contains(self, substr: Union[bytes, str_scalars], regex: bool = False) -> pd
         """
         if isinstance(substr, bytes):
             substr = substr.decode()
-        if regex:
-            if re.search(substr, ''):
-                # TODO remove once changes from chapel issue #18639 are in arkouda
-                raise ValueError("regex operations with a pattern that matches the empty string are not currently supported")
-            matcher = self._get_matcher(substr, create=False)
-            if matcher is not None:
-                return matcher.get_match(MatchType.SEARCH, self).matched()
-        cmd = "segmentedEfunc"
-        args = "{} {} {} {} {} {} {}".format("contains",
-                                             self.objtype,
-                                             self.entry.name,
-                                             "legacy_placeholder",
-                                             "str",
-                                             regex,
-                                             json.dumps([substr]))
+        if not regex:
+            substr = re.escape(substr)
+        if re.search(substr, ''):
```

Comment:
ah yeah the reason I do this is because I'm checking for patterns that match the empty string which is a subtle distinction from being equal to the empty string

so this also catches cases like `substr='$'` or `substr='|'` which still match the empty string and would cause the server to hang for users running anything less than the latest version of chapel